ansible/roles/test/files/ptftests/advanced-reboot.py

```diff
@@ -1487,17 +1448,44 @@ def extract_no_cpu_replies(self, arr):
         else:
             return non_zero[-1]
 
+    def get_teamd_state(self):
+        stdout, stderr, _ = self.dut_connection.execCommand(
+            'sudo systemctl is-active teamd.service')
+        if stderr:
+            self.fails['dut'].add("Error collecting teamd state. stderr: {}, stdout:{}".format(
+                str(stderr), str(stdout)))
+            raise Exception("Error collecting teamd state. stderr: {}, stdout:{}".format(
+                str(stderr), str(stdout)))
+        if not stdout:
+            self.log('teamd state not returned from DUT')
+            return ''
+
+        teamd_state = stdout[0].strip()
+        return teamd_state
+
+    def wait_until_teamd_goes_down(self):
+        self.log('Waiting for teamd service to go down')
+        teamd_state = self.get_teamd_state()
+        self.log('teamd service state: {}'.format(teamd_state))
+        dut_datetime = self.get_now_time()
+        teamd_shutdown_timeout = 300
+
+        while teamd_state == 'active':
```

Comment:
If traffic will start only when teamd is down - that means that in case of fast-boot the traffic will drop from the very first packet. In that case, it would be unclear whether the traffic drops expectedly due to the reboot operation or unexpectedly, due to any other reason that may have caused it prior to the reboot. Suggest keeping the existing logic to start traffic and introduce the new logic to stop traffic(when the finalizer is inactive and the device was rebooted).

Reply:
For fast-reboot the dataplane impact does NOT start when teamd goes down. That would definitely break the permissible limit of 30s.
Note that for T0 device, graceful restart configuration is enabled. So when bgp is down on T0, the peers do NOT withdraw the routes. Hence, even if teamd (LAGs) and BGP are down, dataplane is still functioning.

For fast-reboot, the dataplane impact starts during syncd startup stage when ASIC reset happens.